libraries/node-core-library/src/Terminal/ITerminalProvider.ts

```diff
@@ -2,13 +2,29 @@
 // See LICENSE in the project root for license information.
 
 /**
+ * Similar to many popular logging packages, terminal providers support a range of message
+ * severities. These severities have built-in formatting defaults in the Terminal object
+ * (warnings are yellow, errors are red, etc.).
+ *
+ * Terminal providers may choose to suppress certain messages based on their severity,
+ * or to route some messages to other providers or not based on severity.
+ *
+ *   Severity  | Purpose
+ *   --------- | -------
+ *   error     | Build errors and fatal issues in rush
+ *   warning   | Not necessarily fatal, but indicate a problem the user should fix
+ *   log       | Informational messages from the rush system
+ *   verbose   | Additional information that may not always be necessary
+ *   debug     | Highest detail level, best used for troubleshooting information
+ *
  * @beta
  */
 export enum TerminalProviderSeverity {
-  log,
-  warning,
   error,
-  verbose
+  warning,
+  log,
+  verbose,
+  debug
 }
```

Comment:
I definitely see why you reordered these, but it causes a breaking API change.

Reply:
The order change here and the `getVerbose()` rename are only nice-to-haves.  I did end up marking the the rush change as MAJOR I believe, but now in hindsight, it probably makes more sense to revert those pieces and then this whole change could be minor.

Reply:
Let's avoid the major bump for now. There are a few other breaking changes we've been talking about, so we can make those then.